### PR TITLE
Add Function to Calculate Angle Error

### PR DESCRIPTION
**Pull request by RavenRandomz**
This calculates the shortest angle error.
Before pulling consider incorporating the function itself into where the math stuff is or saving that refactoring for another day.

### CODE REVIEW

**Comment by RavenRandomz** [comment]
I have added directional functionality now. Counterclockwise is positive while clockwise is negative.

File: Core/Src/subsystems/gimbal.cpp [hunk 1]
```diff
@@ -58,18 +60,10 @@ void act() {
 }
 
 double calculateAngleError(double currAngle, double targetAngle) {
-    // IMPLEMENT THIS
-    /* 
-    Given a current angle value (currAngle, in radians) and a target angle value (targetAngle, in radians):
-        - Find and return the shortest angle error in radians with direction.
```
Comment (evan-krohn) [inline Core/Src/subsystems/gimbal.cpp:64]: We missed this line initially; we've updated the pull request with a fix (and the fix is also slightly more efficient for the common case).

File: Core/Src/subsystems/gimbal.cpp [hunk 2]
```diff
@@ -59,11 +57,11 @@ void act() {
     }
 }
 
-double calculateAngleError(double currAngle, double targetAngle) {
-    /* Assumes currAngle and targetAngle are 0<=a<2pi */
+double calculateAngularPath(double currAngle, double targetAngle) {
+	//Positive is counterclockwise while Negative is clockwise
     double angleDelta = currAngle - targetAngle;
 	
-	if (fabs(angleDelta <= PI))
```
Comment (evan-krohn) [inline Core/Src/subsystems/gimbal.cpp]: Minor thing; Tabbing is inconsistent

File: Core/Src/subsystems/gimbal.cpp [hunk 3]
```diff
@@ -59,11 +57,11 @@ void act() {
     }
 }
 
-double calculateAngleError(double currAngle, double targetAngle) {
-    /* Assumes currAngle and targetAngle are 0<=a<2pi */
```
Comment (evan-krohn) [inline Core/Src/subsystems/gimbal.cpp]: I think it makes sense to leave this in as someone quickly glancing at the function might not realize it.

File: Core/Src/subsystems/gimbal.cpp [hunk 4]
```diff
@@ -57,19 +57,22 @@ void act() {
...
-        - Find and return the shortest angle error in radians with direction.
-    (hint hint use trig functions)
-
-    EX: If our current angle is 359 degrees, and the target angle is 2 degrees, this function
-        should return 3 degrees, as the shortest path from the current angle to the target angle
-        will be 3 degrees clockwise.
-        (the example was in degrees, please use radians)
-    */
-    return 0;
+double calculateAngularPath(double currAngle, double targetAngle) {
```
Comment (evan-krohn) [inline Core/Src/subsystems/gimbal.cpp]: Accidentally renamed this? Should be calculateAngleError

File: Core/Src/subsystems/gimbal.cpp [hunk 5]
```diff
@@ -57,19 +57,22 @@ void act() {
...
+	//Positive is counterclockwise while Negative is clockwise
+    double angleDelta = currAngle - targetAngle;
+	
+	if (fabs(angleDelta <= PI)) //Check if within simple case of within 180 degrees
```
Comment (evan-krohn) [inline Core/Src/subsystems/gimbal.cpp]: Might have the parentheses wrong for fabs, right now it doesn't make sense to me

File: Core/Src/subsystems/gimbal.cpp (hunk 3 above)
Comment (evan-krohn) [reply Core/Src/subsystems/gimbal.cpp]: Resolved as per design change

**Review by bid-p — APPROVED** [review]
Looks good, make sure to not change any existing function names unless necessary, and if so change the corresponding function declaration in the header file as well. Code will be tested post-merge.

**Pull request merged**